### ISSUE TITLE
`checkoutWithRequest:fromViewController:completion:` should use NSError default keys to provide concise error info

### ISSUE THREAD

**Issue by alvaromb** [sumup/sumup-ios-sdk#11]
As per https://developer.apple.com/library/ios/documentation/Cocoa/Reference/Foundation/Classes/NSError_Class/index.html#//apple_ref/occ/cl/NSError, `NSError` has a default way to handle application errors with localized description keys. However, when performing a checkout with `checkoutWithRequest:fromViewController:completion:` that causes an error, `localizedFailureReason`, `localizedRecoverySuggestion` and `localizedRecoveryOptions` return `nil`.

Inside the error's `userInfo` there is useful information, this is the console output of a `po [error userInfo]` lldb command.

<img width="693" alt="screen shot 2016-04-04 at 12 14 42" src="https://cloud.githubusercontent.com/assets/289640/14245040/d69a2ba0-fa5e-11e5-84dc-f6a3ff4e3d4f.png">

It's a bit tricky to access to the `message` key, which contains relevant localized information about the error. 

<img width="676" alt="screen shot 2016-04-04 at 12 17 32" src="https://cloud.githubusercontent.com/assets/289640/14245125/728a58d2-fa5f-11e5-98e7-4a79d529bb46.png">

Populating the localized keys with the info of the underlying error would be very helpful.

**Comment by mollidor** [comment]
Hi @alvaromb 

Good point. These messages are displayed by the SumUp app indeed. We've thus added the error per the `NSUnderlyingErrorKey`, which is the "recommended standard way to embed NSErrors from underlying calls". However we missed to populate the `localized....`  properties of the error in a standard way in the SDK. 

We will try to add them in a future SDK version, most likely as `NSLocalizedDescriptionKey`.

Regards,
Lukas

**Comment by alvaromb** [comment]
Yes, I got the underlying error using that key. However, depending on your debug schema (or Xcode bugs), sometimes the underlying error is unaccessible through lldb and `NSError`'s `NSUnderlyingErrorKey`.

For future reference, I got the message using the following code:

```objc
NSString *errorMessageKeyPath = [NSString stringWithFormat:@"%@.userInfo.JSONObject.data.message", NSUnderlyingErrorKey];
NSString *message = [[error userInfo] valueForKeyPath:errorMessageKeyPath];
```

**Comment by larssn** [comment]
Can we revisit this issue? Is it still on the table? It's been two years.

**Comment by mollidor** [comment]
Hi @larssn ,

thanks for bringing this up again and sorry for dropping the ball on this issue. We'll revisit it and ping you once we have an update.

Regards,
Lukas

**Comment by leila-sumup** [comment]
Hello, thanks for reporting this and apologies for our considerable lag here.

The iOS SDK has gone through several major updates since this was originally filed in 2016. Can you confirm whether this is still an issue on the latest version? If we don't hear back, we'll close this out. 

Happy to reopen if it comes up again.

**Comment by larssn** [comment]
Yeah I'd imagine something would happen in 10 years. Think it's safe to close.
